client.listDBs is throwing Error: Method is not accessible, on `onEnd`

**Describe the bug**
client.listDBs is throwing Error: Method is not accessible, on `onEnd` 
> Uncaught (in promise) Error: Method is not accessible
    onEnd index.ts:793

**To Reproduce**
I've a repo [here](https://github.com/Tbaut/textile-playground) where I mostly follow what you do in your test https://github.com/textileio/js-threads/blob/37d8ef1c3ec7a0748465b77d7e29f0cd1494274d/packages/client/src/index.spec.ts

The repo is a CRA for TS, so just clone, `npm install` (or `yarn`), then `yarn start`.
I do the auth with KeyInfo, so you'll need a `.env` file at the root with:
```bash
REACT_APP_HUB_KEY="<your_key>"
REACT_APP_HUB_SECRET="<your_secret>"
```

This is the flow of commands: https://github.com/Tbaut/textile-playground/blob/master/src/components/Threadtest.tsx

Watching the browser console, I get:
![image](https://user-images.githubusercontent.com/33178835/86621178-082bc880-bfbe-11ea-996f-1b0100a8eadd.png)

**Expected behavior**
To list the dbs

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser Firefox
 - Version 78

**Additional context**
Note that I tried to run the test in `js-threads` to see if the problem was on my side, but I run into errors that I don't understand. Wanted to make this issue before falling into the rabbit hole.
```bash
npm run test:browser

> @textile/threads@0.0.0 test:browser /home/thib/github/textileio/js-threads
> lerna run --stream test:browser

lerna notice cli v3.22.1
lerna info versioning independent
lerna info Executing command in 9 packages: "npm run test:browser"
@textile/context: > @textile/context@0.6.0 test:browser /home/thib/github/textileio/js-threads/packages/context
@textile/context: > polendina --cleanup --timeout 60 src/**/*.spec.ts --webpack-config ../../webpack.test.js
@textile/threads-store: > @textile/threads-store@0.1.16 test:browser /home/thib/github/textileio/js-threads/packages/store
@textile/threads-store: > polendina --cleanup --timeout 60 src/**/*.spec.ts --webpack-config ../../webpack.test.js
@textile/threads-id: > @textile/threads-id@0.1.10 test:browser /home/thib/github/textileio/js-threads/packages/id
@textile/threads-id: > polendina --cleanup  src/**/*.spec.ts --webpack-config ../../webpack.test.js
@textile/threads-store:   polendina ☞   Setting up output directory: /home/thib/github/textileio/js-threads/packages/store/build ...
@textile/threads-id:   polendina ☞   Setting up output directory: /home/thib/github/textileio/js-threads/packages/id/build ...
@textile/context:   polendina ☞   Setting up output directory: /home/thib/github/textileio/js-threads/packages/context/build ...
@textile/threads-id: [
@textile/threads-id:   './src/index.ts\n' +
@textile/threads-id:     "Module not found: Error: Can't resolve '@consento/sync-randombytes' in '/home/thib/github/textileio/js-threads/packages/id/src'\n" +
@textile/threads-id:     "resolve '@consento/sync-randombytes' in '/home/thib/github/textileio/js-threads/packages/id/src'\n" +
@textile/threads-id:     '  Parsed request is a module\n' +
...
```

--- a/README.md
+++ b/README.md
@@ -14,7 +14,7 @@
 
 ## Using Threads
 
-To get started using Threads, check out the [docs site](https://docs.textile.io/) and [API documentation](https://textileio.github.io/js-threads).
+To get started using Threads, check out the [docs site](https://docs.textile.io/).
 
 ## Getting help
 

--- a/lerna.json
+++ b/lerna.json
@@ -1,5 +1,7 @@
 {
-  "packages": ["packages/*"],
+  "packages": [
+    "packages/*"
+  ],
   "version": "independent",
   "command": {
     "version": {

--- a/package.json
+++ b/package.json
@@ -3,7 +3,6 @@
   "version": "0.0.0",
   "description": "Typescript/Javascript implementation of Textile's Threads Protocol",
   "main": "packages/threads",
-  "types": "packages/threads",
   "resolutions": {
     "npm-packlist": "1.1.12"
   },
@@ -22,8 +21,8 @@
     "test:node": "TS_NODE_FILES=true mocha --exit",
     "test:browser": "lerna run --stream test:browser",
     "test:docs": "typedoc --out temp && typescript-docs-verifier --input-files=`find temp -type f -name '**.md'`",
+    "textile": "lerna exec -- npx ncu -u '/^@textile/.*$/'",
     "textile:check": "lerna exec -- npx ncu '/^@textile/.*$/'",
-    "textile:update": "lerna exec -- npx ncu -u '/^@textile/.*$/'",
     "docs": "typedoc --plugin typedoc-plugin-lerna-packages && touch ./docs/.nojekyll",
     "ci": "if [ $(npm run textile:check | grep '@textile/' | wc -l) == 0 ]; then echo 'ok'; else exit 1; fi;"
   },

--- a/packages/client/package.json
+++ b/packages/client/package.json
@@ -29,6 +29,7 @@
     "@textile/threads-client-grpc": "^1.0.2",
     "@textile/threads-id": "^0.4.0-alpha.3",
     "@types/to-json-schema": "^0.2.0",
+    "fastestsmallesttextencoderdecoder": "^1.0.22",
     "to-json-schema": "^0.2.5"
   },
   "gitHead": "1c550ec78eab0368d5c758a7068e529c45918729"

--- a/packages/client/src/index.ts
+++ b/packages/client/src/index.ts
@@ -437,18 +437,19 @@ export class Client {
 
   /**
    * Lists all known DBs.
-   * @remarks this API is blocked on the Hub. Use `listThreads` when importing Client
-   * from `@textile/hub` as an alternative.
    */
   public listDBs(): Promise<
-    Record<string, pb.GetDBInfoReply.AsObject | undefined>
+    Record<string, { id: string; name?: string } | undefined>
   > {
     const req = new pb.ListDBsRequest()
     return this.unary(API.ListDBs, req, (res: pb.ListDBsReply) => {
-      const dbs: Record<string, pb.GetDBInfoReply.AsObject | undefined> = {}
+      const dbs: Record<string, { id: string; name?: string } | undefined> = {}
       for (const db of res.getDbsList()) {
         const id = ThreadID.fromBytes(db.getDbid_asU8()).toString()
-        dbs[id] = db.getInfo()?.toObject()
+        dbs[id] = {
+          id,
+          name: db.getInfo()?.getName(),
+        }
       }
       return dbs
     })

--- a/packages/client/src/index.ts
+++ b/packages/client/src/index.ts
@@ -21,6 +21,7 @@ import {
   APIListen,
 } from "@textile/threads-client-grpc/threads_pb_service"
 import { ThreadID } from "@textile/threads-id"
+import "fastestsmallesttextencoderdecoder"
 import toJsonSchema, { JSONSchema3or4 } from "to-json-schema"
 import {
   ComparisonJSON,

--- a/packages/client/src/index.ts
+++ b/packages/client/src/index.ts
@@ -108,7 +108,7 @@ export function getFunctionBody(
  */
 export interface CollectionConfig<W = any, R = W> {
   name: string
-  schema?: JSONSchema3or4 | any // Union type to indicate that JSONSchema is preferred but any works
+  schema?: JSONSchema3or4 | any // JSONSchema is preferred but any works
   indexes?: pb.Index.AsObject[]
   writeValidator?:
     | ((writer: string, event: Event, instance: W) => boolean)

--- a/packages/context/src/index.ts
+++ b/packages/context/src/index.ts
@@ -243,8 +243,7 @@ export class Context implements ContextInterface {
 
   withContext(value?: ContextInterface): this {
     if (value === undefined) return this
-    // Spread to copy rather than reference
-    this._context = value.toJSON()
+    this._context = { ...this._context, ...value.toJSON() }
     return this
   }
 
@@ -285,7 +284,7 @@ export class Context implements ContextInterface {
    * @see toMetadata for an alternative for gRPC clients that supports auto-renewal.
    */
   toJSON(): Record<string | number, any> {
-    const { ...json } = this._context
+    const json = { ...this._context }
     // If we're expired, throw...
     if (this.isExpired) {
       throw errors.expirationError

--- a/packages/multiaddr/package.json
+++ b/packages/multiaddr/package.json
@@ -20,7 +20,7 @@
   "repository": "github:textileio/js-threads",
   "dependencies": {
     "@textile/threads-id": "^0.4.0-alpha.3",
-    "multiaddr": "^8.1.1",
+    "multiaddr": "^8.1.2",
     "varint": "^6.0.0"
   },
   "gitHead": "1c550ec78eab0368d5c758a7068e529c45918729"

--- a/packages/security/package.json
+++ b/packages/security/package.json
@@ -26,6 +26,7 @@
   "dependencies": {
     "@consento/sync-randombytes": "^1.0.5",
     "fast-sha256": "^1.3.0",
+    "fastestsmallesttextencoderdecoder": "^1.0.22",
     "multibase": "^3.1.0"
   },
   "gitHead": "1c550ec78eab0368d5c758a7068e529c45918729"

--- a/packages/security/src/index.ts
+++ b/packages/security/src/index.ts
@@ -6,6 +6,7 @@
  * @packageDocumentation
  */
 import { HMAC } from "fast-sha256"
+import "fastestsmallesttextencoderdecoder"
 import multibase from "multibase"
 
 const encoder = new TextEncoder()

